Sources/ComposableArchitecture/Effects/Cancellation.swift

```diff
@@ -58,37 +39,32 @@ extension Effect {
     return .deferred { () -> SignalProducer<Value, Error> in
       let subject = Signal<Value, Error>.pipe()
 
-      var disposable: Disposable?
       var values: [Value] = []
       var isCaching = true
 
       cancellablesLock.sync {
         if cancelInFlight {
-          cancellationCancellables[id]?.forEach { cancellable in cancellable.dispose() }
+          cancellationCancellables[id]?.dispose()
           cancellationCancellables[id] = nil
         }
 
-        disposable =
-          self
+        let disposable = self
           .on {
             guard isCaching else { return }
             values.append($0)
           }
           .start(subject.input)
 
-        AnyDisposable {
-          disposable?.dispose()
-          subject.input.sendCompleted()
-        }
-        .store(in: &cancellationCancellables[id, default: []])
+          disposable.add(to: &cancellationCancellables[id, default: CompositeDisposable()])
```

Comment:
Refactor all this logic to use `CompositeDisposable` instead of an array of `Disposable`.